src/structures/Message.js

```diff
@@ -127,25 +127,27 @@ class Message extends Base {
      * lost if re-fetched</warn>
      * @type {?string}
      */
-    this.nonce = 'nonce' in data ? data.nonce : null;
+    this.nonce = partial ? this.nonce : 'nonce' in data ? data.nonce : null;
 
     /**
      * A list of embeds in the message - e.g. YouTube Player
      * @type {MessageEmbed[]}
      */
-    this.embeds = data.embeds?.map(e => new Embed(e, true)) ?? [];
+    this.embeds = data.embeds?.map(e => new Embed(e, true)) ?? (partial ? this.embeds.slice() : []);
 
     /**
      * A list of MessageActionRows in the message
      * @type {MessageActionRow[]}
      */
-    this.components = data.components?.map(c => BaseMessageComponent.create(c, this.client)) ?? [];
+    this.components =
+      data.components?.map(c => BaseMessageComponent.create(c, this.client)) ??
+      (partial ? this.components.slice() : []);
 
     /**
      * A collection of attachments in the message - e.g. Pictures - mapped by their ids
      * @type {Collection<Snowflake, MessageAttachment>}
      */
-    this.attachments = new Collection();
+    this.attachments = partial && this.attachments ? new Collection(this.attachments) : new Collection();
```

Comment:
This doesn't seem to do the same thing anymore?

Previously, if `attachments` was present, it'd clear the collection and fill it in with the new attachments, otherwise cloning the existing attachments.

Now, if we get a partial update, we clone the previous attachments and append to them / overwrite some of them.

Am I reading the code wrong?

Reply:
Nope, I wrote it wrong, supposed to be !data.attachments 🤦